lib/src/git_backend.rs

```diff
@@ -420,6 +423,20 @@ fn gix_open_opts_from_settings(settings: &UserSettings) -> gix::open::Options {
         .open_path_as_is(true)
 }
 
+fn root_tree_from_header(git_commit: &CommitRef) -> Option<MergedTreeId> {
+    for (key, value) in &git_commit.extra_headers {
+        if *key == JJ_TREES_COMMIT_HEADER {
+            let tree_ids: MergeBuilder<TreeId> = String::from_utf8_lossy(value.as_ref())
```

Comment:
Maybe better to use `str::from_utf8()` and handle error.

It might be also good to check the hex length here to avoid panic later.

Reply:
Done. I also added a check for the number of trees. This function changed quite a bit, so PTAL.